### PR TITLE
refactor(io): cleanup `BufferFullError` and `PartialReadError` logic

### PR DESCRIPTION
**Pull request by iuioiua**

### CODE REVIEW

File: io/buf_reader.ts [hunk 1]
```diff
@@ -14,10 +14,10 @@ const LF = "\n".charCodeAt(0);
  * @deprecated This will be removed in 1.0.0. Use the {@link https://developer.mozilla.org/en-US/docs/Web/API/Streams_API | Web Streams API} instead.
  */
 export class BufferFullError extends Error {
-  override name = "BufferFullError";
```
Comment (iuioiua) [inline io/buf_reader.ts:17]: Removes the need to document this property.

File: io/buf_reader.ts [hunk 2]
```diff
@@ -26,10 +26,12 @@ export class BufferFullError extends Error {
  * @deprecated This will be removed in 1.0.0. Use the {@link https://developer.mozilla.org/en-US/docs/Web/API/Streams_API | Web Streams API} instead.
  */
 export class PartialReadError extends Error {
-  override name = "PartialReadError";
```
Comment (iuioiua) [inline io/buf_reader.ts:29]: Again, removes the need to document this property.

File: io/buf_reader.ts [hunk 3]
```diff
@@ -26,10 +26,12 @@ export class BufferFullError extends Error {
...
-  partial?: Uint8Array;
-  constructor() {
+  partial: Uint8Array;
+
+  constructor(partial: Uint8Array) {
```
Comment (iuioiua) [inline io/buf_reader.ts:31]: Previously, this property would be set within `try`/`catch` blocks. It should instead just be set during construction (see below).

**Review by kt3k — APPROVED** [review]
LGTM

**Pull request merged**